libraries/Microsoft.Bot.Builder.LUIS/LuisRecognizerMiddleware.cs

```diff
@@ -34,10 +38,26 @@ public LuisRecognizerMiddleware(ILuisModel luisModel, ILuisRecognizerOptions lui
             if (context.Activity.Type == ActivityTypes.Message)
             {
                 var utterance = context.Activity.AsMessageActivity().Text;
-                var result = await _luisRecognizer.Recognize(utterance, CancellationToken.None).ConfigureAwait(false);
-                context.Services.Add(LuisRecognizerResultKey, result);
+                var result = await _luisRecognizer.CallAndRecognize(utterance, CancellationToken.None).ConfigureAwait(false);
+                context.Services.Add(LuisRecognizerResultKey, result.recognizerResult);
+
+                var traceActivity = Activity.CreateEventActivity();
+                traceActivity.Name = LuisTraceEventName;
+                traceActivity.Value = new LuisTraceInfo
+                {
+                    RecognizerResult = result.recognizerResult,
+                    LuisModel = RemoveSensitiveData(_luisModel),
+                    LuisOptions = _luisOptions,
+                    LuisResult = result.luisResult
+                };
+                await context.SendActivity(traceActivity).ConfigureAwait(false);
```

Comment:
Seems like this should be passed to transcript logging middleware. The middleware would decide where to log it. This might be ok for the short term.